src/test/java/com/exasol/glue/FilterConverterTest.java

```diff
@@ -0,0 +1,232 @@
+package com.exasol.glue;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+import com.exasol.sql.expression.BooleanExpression;
+import com.exasol.sql.expression.ValueExpression;
+import com.exasol.sql.expression.rendering.ValueExpressionRenderer;
+import com.exasol.sql.rendering.StringRendererConfig;
+
+import org.apache.spark.sql.sources.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+class FilterConverterTest {
+    private final FilterConverter filterConverter = new FilterConverter();
+
+    private String render(final Filter... filters) {
+        final Optional<BooleanExpression> andExpression = filterConverter.convert(filters);
+        if (andExpression.isPresent()) {
+            final StringRendererConfig rendererConfig = StringRendererConfig.builder().quoteIdentifiers(true).build();
+            final ValueExpressionRenderer renderer = new ValueExpressionRenderer(rendererConfig);
+            ((ValueExpression) andExpression.get()).accept(renderer);
+            return renderer.render();
+        } else {
+            return "";
+        }
+    }
+
+    @Test
+    void testEmptyFilters() {
+        assertThat(render(), equalTo(""));
+    }
+
+    @Test
+    void testNullAndEmptyFilters() {
+        assertAll(() -> assertThat(filterConverter.convert(null), equalTo(Optional.empty())),
+                () -> assertThat(filterConverter.convert(new Filter[] {}), equalTo(Optional.empty())));
+    }
+
+    @Test
+    void testEmptyOptionalIfUnknownFilter() {
+        assertThat(filterConverter.convert(new Filter[] { new EqualNullSafe("c1", "nullValue") }),
+                equalTo(Optional.empty()));
+    }
+
+    @Test
+    void testEmptyOptionalIfOneOfFiltersUnknown() {
```

Comment:
Could you use a parametericed test / multiple tests? That would show clearer which case caused the fail

Reply:
I think assertAll is usually better used for checking multiple aspects of one result